### ISSUE TITLE
valid-types foails for imported typedefs

### ISSUE THREAD

**Issue by l0ner** [gajus/eslint-plugin-jsdoc#1093]
## Expected behavior

When importing types defined in another file using this one liner:

```javascript
/** @typedef {import('../types/foo.js').Foo} Foo */
```

(as suggested in https://github.com/jsdoc/jsdoc/issues/1645#issuecomment-653110277)

eslint complains about syntax error

```
/src/foo/file.js
  6:1  warning  Syntax error in type: import('../types/sometype.js').SomeType    jsdoc/valid-types
```

Even whan the type gets imported correctly.

## Actual behavior

ESlint should not complain about this kind of typedefs.

## ESLint Config

```js
{
	"root": true,
	"env": {
		"commonjs": false,
		"es2021": true,
		"node": true
	},
	"plugins": [
		"jsdoc",
		"eslint-comments"
	],
	"extends": [
		"eslint:recommended",
		"plugin:jsdoc/recommended"
	],
	"parserOptions": {
		"ecmaVersion": 13,
		"sourceType": "module"
	}
}

```

## ESLint sample

File `types/foo.js`

```js
/**
 * My Type
 * @typedef {object} Foo
 * @property {string} Bar - This is bar
 */
```

File `index.js`

```js
/** @typedef {import('../types/foo.js').Foo} Foo */
```

## Environment

- Node version: v18.16.0
- ESLint version v8.41.0
- `eslint-plugin-jsdoc` version: 44.2.5

**Comment by brettz9** [comment]
I'm not able to replicate in the current version of `eslint-plugin-jsdoc` (the version you specified). Do you have any additional settings you are not showing like `settings.jsdoc.mode` set to `jsdoc`?

We may need a minimal test repo so we can reproduce the issue.

**Comment by brettz9** [comment]
Closing but feel free to comment further as needed.

**Comment by l0ner** [comment]
So apparentely i had the version of eslint-plugin-jsdoc i specified installed globally, while the project i was working on was installing locally an older version. VSCode somehow preferred that local older version over the global one. Updated my packages.json with the lastest and now it works fine.
